### PR TITLE
Fix parseApi not accepting versions that don't start with 'v'

### LINKED ISSUE #394: Custom Resources List is Never Loading if version does not match regex "v.*?"
**To Reproduce**
1. Go to 'Custom Resources → Group → Your CRD'
2. Click on selected CRD
3. Wait until list of CRs will load
4. It will never load

**Expected behavior**
Expected to see list of CRs for CRD

**Screenshots**
Lens 2020-05-29 19-45-06

**Environment (please complete the following information):**
- Lens Version: 3.4.0
- OS: OSX
- Installation method: OSX Official Image
 
**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
/Applications/Lens.app/Contents/MacOS/Lens
info: Checking for update
info: Update for version 3.4.0 is not available (latest version: 3.4.0, downgrade is disallowed).
warn: PrometheusLens: failed to list services: [object Object]
warn: PrometheusLens: failed to list services: [object Object]
info: using operator as prometheus provider
info: using operator as prometheus provider
```

Thank you,

### PR DESCRIPTION
**Pull request by Nokel81**
This PR fixes parseApi not accepting versions that don't start with 'v'.

fixes #394

### CODE REVIEW

File: src/main/prometheus/lens.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export class PrometheusLens implements PrometheusProvider {
         port: service.spec.ports[0].port
       }
     } catch(error) {
-      logger.warn(`PrometheusLens: failed to list services: ${error.toString()}`)
+      logger.warn(`PrometheusLens: failed to list services: ${error.response.body.message}`)
```
Comment (nevalla) [inline src/main/prometheus/lens.ts:21]: This is a bit out of scope of this PR, but looking good :)

File: dashboard/client/api/kube-api.ts [hunk 2]
```diff
@@ -32,20 +32,31 @@ export interface IKubeApiLinkRef {
...
 
-  static parseApi(apiPath = "") {
+  static parseApi(apiPath = ""): IKubeApiLinkBase {
     apiPath = new URL(apiPath, location.origin).pathname;
     const [, apiPrefix, apiGroup = "", apiVersion, namespace, resource, name] = apiPath.match(KubeApi.matcher) || [];
     const apiVersionWithGroup = [apiGroup, apiVersion].filter(v => v).join("/");
     const apiBase = [apiPrefix, apiGroup, apiVersion, resource].filter(v => v).join("/");
+
+    if (!apiBase) {
+      throw new Error(`invalid apiPath: ${apiPath}`)
```
Comment (nevalla) [inline dashboard/client/api/kube-api.ts:51]: Did you test how the dashboard is handling this error?

**Review by nevalla — APPROVED** [review]
Did not test the actual behaviour, but the code itself looks good.

**Comment by nevalla** [comment]
@jakolehm Please could you restart the failed CICD build. I don't see any reason why this PR has broken that.

**Comment by nevalla** [comment]
> I don't see any reason why this PR has broken that.

🤔 possibly babel upgrade has broken this.

**Comment by Nokel81** [comment]
Maybe, but if you look at the first type error then it is clearly this bug (https://github.com/mui-org/material-ui/issues/20191) from `material-ui`.

**Comment by Nokel81** [comment]
Would commenting the line with the issue be a good temporary fix?

**Comment by nevalla** [comment]
> Would commenting the line with the issue be a good temporary fix?

What line do you mean?

File: dashboard/client/api/kube-api.ts (hunk 2 above)
Comment (Nokel81) [reply dashboard/client/api/kube-api.ts:51]: I did, the rest of the dashboard works even if this throws.

**Comment by Nokel81** [comment]
`dashboard/client/components/+workloads-deployments/deployment-scale-dialog.tsx:99`

**Comment by Nokel81** [comment]
@nevalla I have fixed the type errors

File: dashboard/client/components/+workloads-deployments/deployment-scale-dialog.tsx [hunk 3]
```diff
@@ -63,8 +63,12 @@ export class DeploymentScaleDialog extends Component<Props> {
     this.ready = false;
   }
 
-  onChange = (evt: React.ChangeEvent, value: number) => {
-    this.desiredReplicas = value;
+  onChange = (evt: React.ChangeEvent, value: number|number[]) => {
+    if (typeof value == "number") {
+      this.desiredReplicas = value;
+    } else if (value.length > 0) {
+      this.desiredReplicas = value.pop();
```
Comment (Nokel81) [inline dashboard/client/components/+workloads-deployments/deployment-scale-dialog.tsx]: This is technically a functional change I think

**Comment by Nokel81** [comment]
A lot of them seem to be cause by `ChartData` not being the same type as what is used as in `ChartDataProps`.

There is a different method of fixing them, that is changing the type of `ChartProps.data` in `dashboard/client/components/chart/chart.tsx`

File: dashboard/client/api/kube-api.ts (hunk 2 above)
Comment (nevalla) [reply dashboard/client/api/kube-api.ts:51]: Good!

**Pull request merged**